### ISSUE TITLE
Give example of shop name format (improve docs)

### ISSUE THREAD

**Issue by marcospgp** [MONEI/Shopify-api-node#164]
The only example of a shop name in the documentation is "your-shop-name". This leaves some ambiguity as to the format of the name, which could be "myshopifystore.shopify.com" or simply "myshopifystore".

**Comment by lpinca** [comment]
If the full domain name was required it would have been `your-shop-name.shopify.com` :)
Happy to accept a PR to clarify this.

**Comment by Freundschaft** [comment]
to be honest I find it a little confusing that this API uses just the shop name instead of the full URL since in the shopify docs, everywhere the URL is used, which means we first have to strip myshopify.com from the url before attaching to this API. is there a specific reason why the full URL isnt used?

**Comment by lpinca** [comment]
@Freundschaft can you show some of the examples that use the domain name instead of the shop name?
The [OAuth guide](https://help.shopify.com/api/getting-started/authentication/oauth#step-2-ask-for-permission) for example uses the shop name, not the domain name.

Anyway this can be changed to make it work with the domain and maintain backward compatibility.

```diff
diff --git a/index.js b/index.js
index 072c1e8..2223b4d 100644
--- a/index.js
+++ b/index.js
@@ -50,7 +50,7 @@ function Shopify(options) {
 
   this.baseUrl = {
     auth: !options.accessToken && `${options.apiKey}:${options.password}`,
-    hostname: `${options.shopName}.myshopify.com`,
+    hostname: `${options.shopName.split('.')[0]}.myshopify.com`,
     protocol: 'https:'
   };

```

**Comment by Freundschaft** [comment]
In my case I'm using the embedded SDK, and shopify is loading the iframe with the shop query parameter as shopname.myshopify.com (https://github.com/Shopify/polaris/blob/master/examples/react-embedded/src/App.js#L6)

also in the shopname info I get the full shopify domain https://help.shopify.com/api/reference/shop (myshopify_domain)

and in verification I'll always get the myshopify.com passed along (https://help.shopify.com/api/getting-started/authentication/oauth#verification)

for all of these I need to remove the myshopify.com

**Comment by lpinca** [comment]
> I think enabling the usage of both would be the smartest idea

Yes I agree as it removes all problems. Will fix this.

**Comment by Freundschaft** [comment]
Thanks!

**Comment by marcospgp** [comment]
```
// Remove '.myshopify.com' suffix
  const pieces = store.url.split('.');
  store.name = pieces.slice(0, pieces.length - 2).join('.');
```

**Comment by lpinca** [comment]
I've just published `shopify-api-node@2.11.0`. It accepts both the shop name and the shop's 'myshopify.com' domain.
